### PR TITLE
feat: support variable & type having same name

### PR DESCRIPTION
**Pull request by moroine**
In Typescript you can have variable & type having the same name but it's not supported by FlowJs.

There is certainly more use cases, but I'm working incrementally on mongodb type delcaration. Will add them later if I encounter the issue.

https://flow.org/try/#0PTAEAEDMBsHsHcBQiAmBTAxtAhgJzaALawoCu0BA5MQHYDmJARpaAN6KiejpZ4FoAPAA6xcAF1AA3PKACCpMQAsAspkXYaASwDOhAFxsOXY8oDyAOQDipgCIAhAPqyA6gGUDAIjNXbdgLQurh4ANEbGnN7W9g4AwgBKnpG+fvEhYaAAvsjGPDj4oGIAnkIE8kqqGOpauqAAvIbhXEnRgYkWUf6BaY0R7b6xCaBeffYpcd1cWRkA3IhAA

### CODE REVIEW

**Comment by orta** [comment]
👍🏻 let me know when you've got enough stuff that you want a release

**Comment by moroine** [comment]
@orta I think we're good to go. I didn't got time to finish this PR to support fully mongo. Here is the output I have https://gist.github.com/moroine/d5e7479469d74953c6f26a9e39d9e7d0

I still have some issues:
1. import from nodejs libs defined in flow internally doesn't have the same name as per TypeScript. We could maintain a mapping 🤔 
2. importing type from external modules is not allowed in flow libs... Somehow we should recreate import types within the file
3. Some TypeScript helper like `Exclude` & `Pick` 
4. What do you think about increasing minimum flowjs version output by the flowgen. Looks like flow is adding new helpers like `$Partial`. Maybe in few months, we could generate a new flowgen major version 🤔

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "@babel/highlight": "^7.10.4",
     "commander": "^6.1.0",
     "lodash": "^4.17.20",
+    "mongodb": "^4.1.3",
```
Comment (orta) [inline package.json:21]: note to self this need switching to devDeps

**Comment by orta** [comment]
4: This makes sense, we can do that 👍🏻

**Comment by orta** [comment]
Shipped 👍🏻

**Pull request merged**